packages/playground/remote/src/lib/worker-thread.ts

```diff
@@ -186,6 +186,86 @@ export class PlaygroundWorkerEndpoint extends PHPWorker {
 	async replayFSJournal(events: FilesystemOperation[]) {
 		return replayFSJournal(this.__internal_getPHP()!, events);
 	}
+
+	async downloadWordPressAssets() {
+		await downloadWordPressAssets(this.__internal_getPHP()!);
+	}
+}
+
+async function downloadWordPressAssets(php: PHP) {
+	if (!php.requestHandler) {
+		logger.warn('No PHP request handler available');
+		return;
+	}
+
+	try {
+		const remoteAssetListPath = joinPaths(
+			php.requestHandler.documentRoot,
+			'wordpress-remote-asset-paths'
+		);
+		// If the remote asset list is not available this means that the WordPress static assets are already downloaded.
+		if (!php.fileExists(remoteAssetListPath)) {
+			return;
+		}
+		const wpVersion = await getLoadedWordPressVersion(php.requestHandler);
+		const staticAssetsDirectory =
+			wpVersionToStaticAssetsDirectory(wpVersion);
+		if (!staticAssetsDirectory) {
+			return;
+		}
+		const response = await fetch(
+			[
+				wordPressSiteUrl,
+				staticAssetsDirectory,
+				'wordpress-static.zip',
+			].join('/')
+		);
+
+		if (!response.ok) {
+			logger.warn('Failed to download WordPress assets');
+			return;
+		}
+
+		const zipPath = '/tmp/wordpress-static-assets.zip';
+		await php.writeFile(
+			zipPath,
+			new Uint8Array(await response.arrayBuffer())
+		);
+		await php.run({
+			code: `<?php
+				$document_root = ${phpVar(php.requestHandler.documentRoot)};
+				$zip_path = ${phpVar(zipPath)};
+				$zip = new ZipArchive;
+				$res = $zip->open($zip_path);
+				if ($res !== TRUE) {
+					return;
+				}
+				for ($i = 0; $i < $zip->numFiles; $i++) {
+					$filename = $zip->getNameIndex($i);
+					$extractPath = str_replace('wordpress-static', $document_root, $filename);
+					// Create directories if they don't exist
+					if (substr($filename, -1) === '/') {
+						if (is_dir($extractPath)) {
+							continue;
+						}
+						mkdir($extractPath, 0777, true);
+					} else {
+						// Extract files
+						copy("zip://$zip_path#".$filename, $extractPath);
+					}
+				}
+				$zip->close();
+			`,
+		});
+		if (await php.fileExists(zipPath)) {
+			await php.unlink(zipPath);
+		}
+		if (await php.fileExists(remoteAssetListPath)) {
+			await php.unlink(remoteAssetListPath);
```

Comment:
AFAICT, deleting this file is going to cause it to be auto-backfilled during the next worker thread initialization here:
https://github.com/WordPress/wordpress-playground/blob/40dca15bf2d5c21c2ee57a878940cb6f7f7851f0/packages/playground/remote/src/lib/worker-thread.ts#L277-L302

I think it may be good to use a more explicit indicator than a missing remote assets listing. But either way, we don't want to be re-requesting the remote asset listing.

If we prefer to use the remote asset listing file as an indicator of the need to download static assets, perhaps we could do one of the following:
- modify the file to reflect only the remote assets that aren't downloaded
- simply empty the file once all remote assets are created locally

Then, if the file exists but is empty, we could assume no further download is required.

Reply:
Emptying the file makes the most sense to me so I implemented it in [de206a1](https://github.com/WordPress/wordpress-playground/pull/1532/commits/de206a14c7251a2fe246e080c8517318d5e242a4).

If we would actively use this file it would make sense because after download there wouldn't be any remote files.